src/modules/form/components/DynamicField.tsx

```diff
@@ -233,6 +234,24 @@ const DynamicField: React.FC<DynamicFieldProps> = ({
           </InputContainer>
         );
 
+      if (item.component === Component.Signature) {
+        const initialValue = {
+          signature: '',
+          date: new Date(),
+          checked: false,
+        };
+        return (
+          <Signature
+            id={linkId}
+            value={value || initialValue}
+            onChange={onChangeValue}
+            whoseSignature={item.text || 'Signature'}
+            {...inputProps}
+            {...commonInputProps}
```

Comment:
Many of the other components returned in this big switch statement pass in `sx` having to do with width; I didn't insert it here because I couldn't see if it was making a difference, but I didn't test on a ton of different screen widths; should it be added here too?